openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1157 @@
+                val = [cell for cell in \
+                       self.geometry.get_all_cells().values() \
+                       if cell.id == int(val)][0]
+            else:
+                check_value('Cell name exists', val, [cell.name for cell in \
+                                self.geometry.get_all_cells().values()])
+
+                val = [cell for cell in \
+                       self.geometry.get_all_cells().values() \
+                       if cell.name == val][0]
+
+        elif isinstance(val, int):
+            check_value('Cell id exists', val, [cell.id for cell in \
+                                self.geometry.get_all_cells().values()])
+            val = [cell for cell in \
+                       self.geometry.get_all_cells().values() \
+                       if cell.id == val][0]
+        else:
+            ValueError(f'Cell: {val} is not supported')
+
+        return val
+
+    def _model_builder(self, param):
+        """
+        Builds the parametric model to be passed to the
+        :meth:`openmc.search.search_for_keff` method.
+        Callable function which builds a model according to a passed
+        parameter. This function must return an openmc.model.Model object.
+        Parameters
+        ----------
+        param : model parametric variable
+            cell translation coefficient or cell rotation coefficient
+        Returns
+        -------
+        self.model :  openmc.model.Model
+            Openmc parametric model
+        """
+        self._set_cell_attrib(param)
+        # At this stage it not important to reassign the new volume, as the
+        # nuclide densities remain constant. However, if at least one of the cell
+        # materials is set as depletable, the volume change needs to be accounted for
+        # as an increase or reduction of number of atoms, i.e. vector x, before
+        # solving the depletion equations
+        return self.model
+
+    def search_for_keff(self, x, step_index):
+        """
+        Perform the criticality search on the parametric cell coefficient and
+        update materials accordingly.
+        The :meth:`openmc.search.search_for_keff` solution is then set as the
+        new cell attribute.
+        Parameters
+        ----------
+        x : list of numpy.ndarray
+            Total atoms concentrations
+        Returns
+        -------
+        x : list of numpy.ndarray
+            Updated total atoms concentrations
+        """
+        # Get cell attribute from previous iteration
+        val = self._get_cell_attrib()
+        check_type('Cell coeff', val, Real)
+
+        # Update all material densities from concentration vectors
+        #before performing the search_for_keff. This is needed before running
+        # the  transport equations in the search_for_keff algorithm
+        super()._update_materials(x)
+
+        # Calculate new cell attribute
+        root = super()._search_for_keff(val)
```

Comment:
Nit pick but we don't usually see calls to `super` when we're calling a different method from the parent class. It shows up in `__init__` because we want to call the parent classes `__init__` method, but we don't need it all the time. Recommend replacing these `super` calls with `self._update_materials` and `self._search_for_keff`